### PR TITLE
Syntactic Filter

### LINKED ISSUE #25: Baseline: syntax filtering
Filter continuous sequence of statements based on block-statement graph.

### PR DESCRIPTION
**Pull request by aravij**
Fix #25.

Syntactic check is done by traversing block-statement graph.
The statements must be represented as a series of subtrees, with roots being a continuos sequence of statements on same block. All such subtrees must be covered fully with statements from extraction opportunity.

### CODE REVIEW

File: veniq/baselines/semi/filter.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
+from typing import List, Optional, Union
+
+from veniq.ast_framework import AST, ASTNode
+from veniq.ast_framework.block_statement_graph import Block, Statement, build_block_statement_graph
+
+
+ExtractionOpportunity = List[ASTNode]
```
Comment (Vitaly-Protasov) [inline veniq/baselines/semi/filter.py:7]: Could you clarify why you decided to use this one. Also, after that, as a typing, you use only "List[ASTNode]", for example at the line 32. What is the point?

File: test/extract_method_baseline/filtering/test_filtering.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+from veniq.baselines.semi.filter import filter_extraction_opportunities, ExtractionOpportunity
+from veniq.ast_framework import AST, ASTNode, ASTNodeType
+from veniq.utils.ast_builder import build_ast
+
+
+class FilteringTestCase(TestCase):
+    def test_non_continuos_statements_range(self):
+        # lines 9 and 15
+        self._opportunity_test_helper([59, 71], False)
```
Comment (acheshkov) [inline test/extract_method_baseline/filtering/test_filtering.py:13]: I did not catch how the test works. I first thought that `[59, 71]` is extraction lines range, but it's not

Comment (aravij) [reply test/extract_method_baseline/filtering/test_filtering.py:13]: The `ExtractionOpportunity` is a list of ASTNode. Here is ASTNode IDs. They get transformed into ASTNodes in `_create_extraction_opportunity` method. There is no other way to point directly to specific node. Comments in each test help to find specified nodes in source code.

File: veniq/baselines/semi/filter.py (hunk 1 above)
Comment (aravij) [reply veniq/baselines/semi/filter.py:7]: I changed `List[ASTNode]` to `ExtractionOpportunity`

File: test/extract_method_baseline/filtering/test_filtering.py (hunk 2 above)
Comment (acheshkov) [reply test/extract_method_baseline/filtering/test_filtering.py:13]: I recommend redo the test interface and pass line ranges. It's more intuitive and does not depend on ID assignment which we do not control here.

**Comment by aravij** [comment]
> I recommend redo the test interface and pass line ranges. It's more intuitive and does not depend on ID assignment which we do not control here. 

Fixed.

File: test/extract_method_baseline/filtering/test_filtering.py (hunk 2 above)
Comment (aravij) [reply test/extract_method_baseline/filtering/test_filtering.py]: Fixed

**Review by lyriccoder — COMMENTED** [review]
ok

**Review by acheshkov — APPROVED** [review]

**Comment by aravij** [comment]
I will postpone merging this PR till #32

**Pull request merged**